### DISCUSSION TITLE
implementation question / missing docs / test not enabled

### DISCUSSION THREAD

**Discussion by delta1186** [spatie/laravel-tags#372] [Bugs] [answered]
@freekmurze I have come across a few items, but before I created a PR I wanted to get your opinion.  

The following two methods currently are not being explicitly tested.  Would you like for me add some tests?
```
Tag::findFromString(string $name, string $type = null, string $locale = null)
Tag::findFromStringOfAnyType(string $name, string $locale = null)
```

Another question is how these methods were meant to be implemented.  On the docs site you can just call it like so
```$tag = Tag::findFromString('apple')``` and return a tag with no type. The method uses ->first() to return the tag model to the user.  This make sense, since there should not be any duplicate tags.  However, ```Tag::findFromStringOfAnyType``` also returns a tag instance using ->first().  Should this not be a collection of tags using ->get()?

I can fix this and update the docs.

Next, I found a test that was not implemented in TagTest.php, ```public function it_can_find_a_tag_from_a_string()```  It currently does not work, but I can fix it.  Would you like for me to do this too?

Let me know your thoughts.  Would it be okay to just submit all these items in the same PR?

Regards,
Kevin

**Comment by freekmurze** [comment, marked as answer]
Thanks, I think you're correct on all accounts. Could you please create separate PRs for me to review?

**Reply by delta1186** [reply]
I am working on the 3rd PR, but I need to give it some more thought.  When I change ->first to ->get it causes the two tests in the existing suite to fail.  I have track this down to a protected method on HasTags convertToTagsOfAnyType where it can accept an array of tags or strings.  It maps the values into a collection.  It then uses the findFromStringOfAnyType, which now returns a collection of tags, but the map is expecting a single tag.  🤔
